packages/js/src/annotations.ts

```diff
@@ -0,0 +1,56 @@
+import HTTPClient from './httpClient.js';
+
+export namespace annotations {
+
+    export interface Annotation {
+        id: string;
+        type: string;
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
+        endTime?: string;
+    }
+
+    export interface ListingQueryParams {
+        datasets?: string[],
+        start?: string,
+        end?: string
+    }
+
+    export interface CreateRequest {
+        type: string;
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
+        endTime?: string;
+    }
+
+    export interface UpdateRequest {
+        type: string;
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
+        endTime?: string;
```

Comment:
every field is optional here

Reply:
hmmmm, I thought we don't do patch requests, but only PUT requests with the full object. That's why I thought the required fields for creation are required here as well.

Reply:
updated